app/main.py

```diff
@@ -0,0 +1,474 @@
+            },
+        }
+
+        self.api = Api(self.app,
+                       version='0.5',
+                       title='LED API',
+                       description='A simple LED IOT API',
+                       host='api.ashleynewton.net',
+                       doc='/docs',
+                       security=['Bearer Auth',
+                                 #    {'OAuth2': ['read', 'write']}
+                                 ],
+                       authorizations=authorizations
+                       )
+
+        CORS(self.app)
+
+    def run(self):
+        """Run flask app."""
+        logging.debug("running Server app")
+        self.app.run(debug=True,
+                     port=80)
+
+
+server = Server()
+APP = server.app
+API = server.api
+CORS(APP)
+
+# Default device settings.
+DEFAULT = {
+    'onState': 'true',
+    'brightness': '255',
+    'name': 'Default'
+}
+
+
+# Single Device Data Model
+DEVICE = API.model('Device', {
+    'onState': fields.String(description='The on/off state',
+                             attribute='onState',
+                             required=False,
+                             default=True),
+    'brightness': fields.String(description='The LED brightness',
+                                attribute='brightness',
+                                min=0,
+                                max=255,
+                                required=False,
+                                default=255),
+    'name': fields.String(description="name of the device",
+                          attribute='name',
+                          required=False,
+                          default='N/A')
+})
+
+# Device List Data Model
+LIST_OF_DEVICES = API.model('ListedDevices', {
+    'id': fields.String(required=True, description='The device ID'),
+    'device': fields.Nested(DEVICE, description='The device'),
+})
+
+HEARTBEAT = API.model('Heartbeat', {
+    'heartbeat': fields.String(required=False, description='Device heartbeat')
+})
+
+
+class Redis(object):
+    """Class for manipulating redis client."""
+
+    def __init__(self):
+        """Initialize Redis Object."""
+        if 'REDIS_HOST' in os.environ:
+            REDIS_HOST = os.environ['REDIS_HOST']
+            if 'REDIS_PASSWORD' in os.environ:
+                REDIS_PASSWORD = os.environ['REDIS_PASSWORD']
+                logging.info("Redis password found %s", REDIS_PASSWORD)
+                self.redis = redis.Redis(
+                    host=REDIS_HOST, port=6379, db=0, password=REDIS_PASSWORD)
+            else:
+                logging.info(
+                    "No redis password found, initializing without auth")
+                self.redis = redis.Redis(host=REDIS_HOST, port=6379, db=0)
+        else:
+            try:
+                raise Exception
+            except Exception:
+                logging.error('REDIS_HOST not set.')
+                sys.exit("!!!Exiting. Please set REDIS_HOST in env & restart.")
```

Comment:
I don't think this exception does anything
```suggestion
            logging.error('REDIS_HOST not set.')
            sys.exit("!!!Exiting. Please set REDIS_HOST in env & restart.")
```

Reply:
Also a good catch. I think I had that in there initially when I was just exploring raising exceptions. Oops. :)